streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1066,9 +1066,12 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
    * type with the specified mapping functions.
    */
   final def mergeWith[R1 <: R, E1 >: E, B, C](that: ZStream[R1, E1, B])(l: A => C, r: B => C): ZStream[R1, E1, C] = {
-    type Loser = Either[Fiber[Nothing, Take[E1, A]], Fiber[Nothing, Take[E1, B]]]
+    type Loser = Either[Fiber[Nothing, Take[E, A]], Fiber[Nothing, Take[E1, B]]]
 
-    def race(left: UIO[Take[E1, A]], right: UIO[Take[E1, B]]): UIO[(Take[E1, C], Loser)] =
+    def race(
```

Comment:
I think we could probably get rid of `Take` entirely here, but we can do that in a followup.